src/workloads/scale/MultiPlanStorm.yml

```diff
@@ -0,0 +1,93 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: >
+  The workload tests the server under a "multi-plan storm" situation. The same query requiring a
+  multi-plan is executed by many threads, each of them triggering a multi-plan. Each of the plans in
+  the multi-plan, on each thread, is index scanning a large number of documents, while maintaining a
+  RecordId set. This causes the memory footprint to increase until the server is eventually OOM
+  killed.
+
+GlobalDefaults:
+  MaxPhases: &MaxPhases 2
+  # The amount of documents here is important, since trial-runs in the query planner "tournament"
+  # will only use ~30% of total document count increasing the number of total documents helps
+  # reproduce the issue.
+  NumDocuments: &NumDocuments 30000000
+  PerfTagDocs: &PerfTagDocs 300
+  RandomTagDocs: &RandomTagDocs
+    ^NumExpr:
+      withExpression: "num_docs - num_perf_tags"
+      andValues: {num_docs: *NumDocuments, num_perf_tags: *PerfTagDocs}
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 6000
+      socketTimeoutMS: 7_200_000  # = 2 hour
+      connectTimeoutMS: 7_200_000
+
+Actors:
+
+# Phases:
+# 1. Insert documents.
+# 2. Generate multi-plan storm.
+
+- Name: InsertData
+  Type: Loader
+  Threads: 10
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Database: &DB test
+        Blocking: None
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1  # Collection name will be Collection0, this is not configurable.
+        DocumentCount: *NumDocuments
+        BatchSize: 1000
+        Document:
+          SpaceId: 5
+          Type: 6
+          Tags:
+            - {^Choose: {from: ['perf', {^FastRandomString: {length: 7}}], weights: [*PerfTagDocs, *RandomTagDocs]}}
+        Indexes:
+          - keys: { SpaceId: 1 }
+          - keys: { Type: 1 }
+          - keys: { Tags: 1 }
+          - keys: { SpaceId: 1, Type: 1 }
+          - keys: { SpaceId: 1, Tags: 1 }
+          - keys: { Type: 1, Tags: 1 }
+          - keys: { SpaceId: 1, Type: 1, Tags: 1 }
+
+- Name: MultiPlanStorm
+  Type: CrudActor
+  Database: *DB
+  Threads: 1000
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: 30 minutes
+        Collection: Collection0
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: { $and:
+              [
+              { SpaceId: { $eq: 5 } },
+              { Type: { $eq: 6 } },
+              { Tags: { $regex: /erf$/, $options: "i"}},
+              { Tags: { $regex: /^per$/, $options: "i"}},
+              { Tags: { $regex: /^perf$/, $options: "i"}},
```

Comment:
```suggestion
              { Tags: { $regex: "/erf$/", $options: "i"}},
              { Tags: { $regex: "/^per$/", $options: "i"}},
              { Tags: { $regex: "/^perf$/", $options: "i"}},
```

is this correct?

Reply:
Correct in what sense? As in properly passed through to the server?

This is what I get from a log line in the server:
```
        "cmd": {
            "find": "Collection0",
            "filter": {
                "$and": [
                    {
                        "SpaceId": {
                            "$eq": 5
                        }
                    },
                    {
                        "Type": {
                            "$eq": 6
                        }
                    },
                    {
                        "Tags": {
                            "$regex": "/erf$/",
                            "$options": "i"
                        }
                    },
                    {
                        "Tags": {
                            "$regex": "/^per$/",
                            "$options": "i"
                        }
                    },
                    {
                        "Tags": {
                            "$regex": "/^perf$/",
                            "$options": "i"
                        }
                    }
                ]
            }
```

Reply:
Done! I will resolve.